### PR TITLE
chore: 10285 split records

### LINKED ISSUE #10285: Split config records based on usage
The Config records `BasicConfig`, `StateConfig`, and `TransactionConfig` are used in `swirlds-common` and `swirlds-platform-core`. The records should be splitted up.

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
**Description**:
Separating records BasicConfig, StateConfig, and TransactionConfig to a platform-core and common counterparts 

**Related issue(s)**:

Fixes #10285

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/BasicConfig.java [hunk 1]
```diff
@@ -72,21 +68,19 @@
...
-        @ConfigProperty(defaultValue = "10") int freezeSecondsAfterStartup,
-        @ConfigProperty(defaultValue = "true") boolean loadKeysFromPfxFiles,
-        @ConfigProperty(defaultValue = "1000") int jvmPauseDetectorSleepMs,
-        @ConfigProperty(defaultValue = "1000") int jvmPauseReportMs,
-        @ConfigProperty(defaultValue = "true") boolean enablePingTrans,
-        @ConfigProperty(defaultValue = "1") long pingTransFreq,
-        @ConfigProperty(defaultValue = "60s") Duration hangingThreadDuration,
-        @ConfigProperty(defaultValue = "data/saved") String emergencyRecoveryFileLoadDir,
-        @ConfigProperty(defaultValue = "0") long genesisFreezeTime) {
+        @ConfigProperty(defaultValue = "1000") int numConnections, // core
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/BasicConfig.java:71]: Did you mean to remove these trailing comments? Ones that are not referenced should be removed

**Review by hendrikebbers — DISMISSED** [review]

**Review by JeffreyDallas — DISMISSED** [review]

**Review by nathanklick — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/BasicConfig.java (hunk 1 above)
Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/BasicConfig.java:71]: Yes, I did; I removed the comments from the other records and missed this one. Thanks for spotting.
Cool, I'll also remove the unreferenced properties 🧹

Comment (mxtartaglia-sl) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/BasicConfig.java]: Done @poulok  
@nathanklick. Please check, given that this introduced a change in settings.txt

**Review by lpetrovic05 — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

File: hedera-node/hedera-app/src/test/resources/signedState/MHS/settingsUsed.txt [hunk 2]
```diff
@@ -32,7 +32,6 @@ The following are all the settings, as modified by settings.txt, but not reflect
            5000 = timeoutSyncClientConnect
            5000 = timeoutServerAcceptConnect
            1000 = deadlockCheckPeriod
-            500 = sleepHeartbeat
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/resources/signedState/MHS/settingsUsed.txt:35]: Is this intentional ?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/test/resources/signedState/MHS/settingsUsed.txt:35]: Seems like it is and is approved by platform.

Comment (mxtartaglia-sl) [reply hedera-node/hedera-app/src/test/resources/signedState/MHS/settingsUsed.txt:35]: It is! I removed unreferenced properties from the record and from the file too.

**Pull request merged**